### ISSUE TITLE
apply to RNNs?

### ISSUE THREAD

**Issue by QihongL** [philipperemy/keras-visualize-activations#4]
Thanks for implementing this! Have you tested your package with recurrent nets? Naively applying your package to an RNN returned some error message and I am trying to understand it... But never mind if this is specifically designed for feedforward nets!

**Comment by philipperemy** [comment]
Hum I've never tried with RNN! Let's try that together :)

**Comment by philipperemy** [comment]
@QihongL Do you think you can work out a PR?

(This objective of the repo is to be generic and work for every possible type of architecture)

**Comment by philipperemy** [comment]
This code works very well for a LSTM.

Can you have a look and see where is your problem?

```
import keras
from keras.layers import Dense, Flatten
from keras.layers.recurrent import LSTM
from keras.models import Sequential

from data import get_mnist_data, num_classes
from read_activations import get_activations

if __name__ == '__main__':
    x_train, _, _, _ = get_mnist_data()

    # (60000, 28, 28, 1) to ((60000, 28, 28)
    # LSTM has (batch, time_steps, input_dim)
    x_train = x_train.squeeze()

    model = Sequential()
    model.add(LSTM(16, input_shape=(28, 28)))
    model.add(Dense(num_classes, activation='softmax'))

    model.compile(loss=keras.losses.categorical_crossentropy,
                  optimizer=keras.optimizers.Adadelta(),
                  metrics=['accuracy'])

    get_activations(model, x_train, print_shape_only=True)
```

Console output:
```
> python3 lstm.py
Using TensorFlow backend.
x_train shape: (60000, 28, 28, 1)
60000 train samples
10000 test samples
----- activations -----
(60000, 16)
(60000, 10)
```

**Comment by QihongL** [comment]
Right, I forgot to update this... I did manage to get it to work and I don't remember what was causing the error. Anyway, your code works for RNNs! Thank you so much! This is super helpful!

**Comment by philipperemy** [comment]
@QihongL happy that it worked for you!!

**Comment by barloccia** [comment]
Hello!
I am in a similar situation. I have a network composed by: CNN for feature extraction + LSTM for sequence classification. The input of the network are 30 frames, and the final output is a class over 4 different ones. I would like to display the feature (extracted before the LSTM) that makes the classification and visualized them over all the frames that have been fed into the lstm. Is that possible?
The code is something like that
`inc = InceptionV3(
    weights='imagenet',
    include_top=False,
    pooling='max',
    input_shape=(shape[0], shape[1], 3)
    )

    for layer in inc.layers:
        layer.trainable = False

    model = Sequential()
    model.add(
            TimeDistributed(inc, input_shape=(chunk_frames, shape[0], shape[1], dataset_.n_channels))
    )
    model.add(Dropout(0.8))
    model.add(LSTM(256, activation='relu', return_sequences=False, kernel_initializer=he_uniform(), bias_initia`
I have doubts since I classify only the time series, not all the input frames.


I really appreciate your work, hope you can clarify my question, 
Bests

**Comment by philipperemy** [comment]
@barloccia can you open a new issue? And provide a full snippet that is easy to run (smallest python code that can reproduce your problem)? Thanks
